sklearn/datasets/__init__.py

```diff
@@ -99,3 +99,50 @@
     "make_spd_matrix",
     "make_swiss_roll",
 ]
+
+
+def __getattr__(name):
+    if name == "load_boston":
+        msg = textwrap.dedent(
+            """
+            `load_boston` has be removed from scikit-learn since version 1.2.
+
+            The Boston housing prices dataset has an ethical problem. You can refer to
+            the documentation of this function for further details.
```

Comment:
the documentation of the function doesn't exist anymore. We need to adapt this message to reflect why it was removed.